pkg/controllers/vdb/revivedb_reconciler.go

```diff
@@ -134,6 +136,35 @@ func (r *ReviveDBReconciler) execCmd(ctx context.Context, initiatorPod types.Nam
 	opts := r.genReviveOpts(initiatorPod, hostList, podNames)
 	r.VRec.Event(r.Vdb, corev1.EventTypeNormal, events.ReviveDBStart, "Starting revive database")
 	start := time.Now()
+	// when preserving DB directory, we need to delete vertica.conf and catalog files using drop_db
+	if vmeta.UseVClusterOps(r.Vdb.Annotations) && vmeta.GetPreserveDBDirectory(r.Vdb.Annotations) {
+		opts := []dropdb.Option{
+			dropdb.WithDBName(r.Vdb.Spec.DBName),
+		}
+		for _, pod := range r.PFacts.Detail {
+			if !pod.GetIsPodRunning() {
+				r.Log.Info("Some pods are not running. Need to requeue revivedb reconciler.", "pod", pod.GetName())
+				return ctrl.Result{Requeue: true}, nil
+			}
+			cmd := []string{"bash", "-c", fmt.Sprintf("ls /%s/%s | grep -E '^v_%s_node[0-9]{4}_catalog$' | sed 's/_catalog$//'",
+				r.Vdb.Spec.Local.GetCatalogPath(), r.Vdb.Spec.DBName, r.Vdb.Spec.DBName)}
+			nodeName, _, err := r.PRunner.ExecInPod(ctx, pod.GetName(), pod.GetExecContainerName(), cmd...)
+			if err != nil {
+				r.Log.Error(err, "Failed to get node name inside a pod", "pod", pod.GetName())
+				return ctrl.Result{}, err
+			}
+			opts = append(opts, dropdb.WithHost(strings.TrimSpace(nodeName), pod.GetPodIP()))
```

Comment:
Would it be better to use pod.vnodeName to get nodeName? Or get the node name from podfacts nodeDetail to be more accurate?